filebeat/input/journald/pkg/journalctl/journalctl.go

```diff
@@ -95,7 +96,20 @@ func Factory(canceller input.Canceler, logger *logp.Logger, binary string, args
 			data, err := reader.ReadBytes('\n')
 			if err != nil {
 				if !errors.Is(err, io.EOF) {
-					logger.Errorf("cannot read from journalctl stdout: %s", err)
+					var logError = false
+					var pathError *fs.PathError
+					if errors.As(err, &pathError) {
+						if pathError.Op == "read" && pathError.Path == "|0" && pathError.Err.Error() == "file already closed" {
```

Comment:
Out of curiosity, what path `"|0"` means?

Reply:
I believe, based on observations, it means `stdout`. I searched a bit for it in the Go source code but did not find many references. It appears a couple of times in the `os` package:
- https://github.com/golang/go/blob/d0631b90a3b0934d0fe223e2d889d785d297f083/src/os/pipe_unix.go#L27
- https://github.com/golang/go/blob/d0631b90a3b0934d0fe223e2d889d785d297f083/src/os/pipe2_unix.go#L21

I did not try reproducing this situation and tracing back how the error gets constructed. Those values for `fs.PathError` are based on observations of what happens. I believe I can add some comments there better explaining the situation.

In short, when reading from journalctl stdout, sometimes journalctl will successfully exit and we're still trying to read from it, sometimes we get a EOF, other times we get this error. I believe it is caused by a timing 'issue' while the process is being shutdown and we're still trying to read. It is harmless and no data is lost, but we have to handle the error.

I ended up adding this condition there to avoid having a log level error on user's deployments that are an expected and well handled situation.

Reply:
Oh, yeah it makes sense now. It is a pipe to stdout, kinda obvious when you reason with each symbol individually, but for some reason it did not click for me at first. TIL. 

A short comment explaining that this is checking specifically for read errors from the journald stdout would be great.